### PR TITLE
Change all references to docker registry to container registry

### PR DESCRIPTION
**Pull request by rhatdan**
Will leave references to distribution spec until there is an OCI Spec we
can point at.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@vrothberg Do you want these to say `container image registry`

**Review by vrothberg — COMMENTED** [review]
LGTM

I am cool with "container registry" (less words). @mtrmac WDYT?

File: internal/iolimits/iolimits.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ const (
 	// `ReadAtMost`.
 	megaByte = 1 << 20
 	// MaxManifestBodySize is the maximum allowed size of a manifest. The limit
-	// of 4 MB aligns with the one of a Docker registry:
+	// of 4 MB aligns with the one of a container registry:
```
Comment (mtrmac) [inline internal/iolimits/iolimits.go:20]: This is a reference to a specific implementation, not the generic concept.

File: manifest/manifest.go [hunk 2]
```diff
@@ -195,7 +195,7 @@ func MatchesDigest(manifest []byte, expectedDigest digest.Digest) (bool, error)
 }
 
 // AddDummyV2S1Signature adds an JWS signature with a temporary key (i.e. useless) to a v2s1 manifest.
-// This is useful to make the manifest acceptable to a Docker Registry (even though nothing needs or wants the JWS signature).
+// This is useful to make the manifest acceptable to a container registry (even though nothing needs or wants the JWS signature).
```
Comment (mtrmac) [inline manifest/manifest.go:198]: This is (mostly() a reference to a specific implementation.

File: openshift/openshift.go [hunk 3]
```diff
@@ -164,7 +164,7 @@ type openshiftImageSource struct {
 	// Values specific to this image
 	sys *types.SystemContext
 	// State
-	docker               types.ImageSource // The Docker Registry endpoint, or nil if not resolved yet
+	docker               types.ImageSource // The container registry endpoint, or nil if not resolved yet
```
Comment (mtrmac) [inline openshift/openshift.go:167]: This is confusing; the OpenShift integrated registry is also a “container registry”, so a “container registry endpoint” is not a distinction.

Refer to the docker/distribution (distribution/distribution? the WIP OCI distribution idea?) protocol here.

Also the other instance.

File: internal/iolimits/iolimits.go (hunk 1 above)
Comment (rhatdan) [reply internal/iolimits/iolimits.go:20]: I am not sure what you means Docker distribution registry?  Docker.io?

File: manifest/manifest.go (hunk 2 above)
Comment (rhatdan) [reply manifest/manifest.go:198]: Should we say older container registry, previous container registry?  Or is the a specific version of the Docker distribution that you want to reference?

File: openshift/openshift.go (hunk 3 above)
Comment (rhatdan) [reply openshift/openshift.go:167]: So you want to say the docker/distribution endpoint?

File: internal/iolimits/iolimits.go (hunk 1 above)
Comment (mtrmac) [reply internal/iolimits/iolimits.go:20]: The line just below links to the implementation in question. (I don’t know what exactly runs `docker.io` but it’s attractive to guess there’s a relationship.)

File: manifest/manifest.go (hunk 2 above)
Comment (mtrmac) [reply manifest/manifest.go:198]: The commit message for this function refers to docker/distribution.

I have no idea whether other implementations accept unsigned v2s1.

File: openshift/openshift.go (hunk 3 above)
Comment (mtrmac) [reply openshift/openshift.go:167]: “docker/distribution API” I guess

Comment (rhatdan) [reply openshift/openshift.go:167]: Fixed

File: internal/iolimits/iolimits.go (hunk 1 above)
Comment (rhatdan) [reply internal/iolimits/iolimits.go]: Removed change.  Not really worth changing some of these.  I wanted to change the ones that show up in users faces.  But just grepped the code for any "docker registry" string.

File: openshift/openshift.go [hunk 4]
```diff
@@ -316,7 +316,7 @@ func (s *openshiftImageSource) ensureImageIsResolved(ctx context.Context) error
 
 type openshiftImageDestination struct {
 	client *openshiftClient
-	docker types.ImageDestination // The Docker Registry endpoint
+	docker types.ImageDestination // The container registry endpoint
```
Comment (mtrmac) [inline openshift/openshift.go:319]: `docker/distribution API` here as well, please.

**Review by mtrmac — COMMENTED** [review]
LGTM after one more change.

Thanks!

**Review by mtrmac — APPROVED** [review]

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**